src/main/java/org/vaadin/addons/tatu/TabSheet.java

```diff
@@ -239,20 +249,58 @@ public void setSelected(String tab) {
         getElement().executeJs("this.setSelectedTab($0)", tab);
     }
 
+    /**
+     * Get selected tab index.
+     *
+     * @return Index of the tab, base 0 or -1 of no selection.
+     */
+    public int getSelectedIndex() {
+        return this.getElement().getProperty("selected", -1);
+    }
     /**
      * Set selected tab using index. This will fire TabChangeEvent. Sheet
      * attached to the tab will be shown.
      * 
      * @param index
      *            Index of the tab, base 0.
      */
-    public void setSelected(int index) {
+    public void setSelectedIndex(int index) {
         if (index < 0) {
             throw new IllegalArgumentException("Index can't be negative");
         }
         getElement().setProperty("selected", index);
     }
 
+    /**
+     * Get selected tab Component.
+     *
+     * @return Selected tab Component or <code>null</code> if no selection.
+     */
+    public Component getSelectedTab() {
+        String id = getSelected();
+        return id == null ? null : this.getComponent(id).orElse(null);
+    }
+    /**
+     * Set selected tab using index. This will fire TabChangeEvent. Sheet
+     * attached to the tab will be shown.
+     *
+     * @param tab
+     *            Component to select.
+     */
+    public void setSelectedTab(Component tab) {
```

Comment:
Note, if the component is used declaratively, and populated in the template, the component instances are not synced to server. Hence in such use case this method does not work, as there are no children. I think there should be a note about this in JavaDoc.

Reply:
Architectural limitation and no way around it, I guess? I'll add a note. Fortunately, dynamic content (together with Java API) likely implies that the content is added from the API. Like in the tests.

Reply:
> Architectural limitation and no way around it, I guess?

Yes. We have couple of similar cases in Vaadin stock components as well. See, e.g. https://github.com/vaadin/flow-components/issues/3838